packages/dashboard/src/components/broadcastsTable.tsx

```diff
@@ -0,0 +1,94 @@
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import { useRouter } from "next/router";
+import React from "react";
+
+import { useAppStore } from "../lib/appStore";
+import { monospaceCell } from "../lib/datagridCells";
+
+interface Row {
+  id: string;
+  name: string;
+  updatedAt: string;
+  triggeredAt: string;
+}
+
+const baseColumn: Partial<GridColDef<Row>> = {
+  flex: 1,
+  sortable: false,
+  filterable: false,
+  renderCell: monospaceCell,
+};
+
+export default function BroadcastsTable() {
+  const router = useRouter();
+  const broadcasts = useAppStore((store) => store.broadcasts);
+
+  const broadcastsRow: Row[] = [];
+
+  broadcasts.forEach((broadcast) => {
+    const row: Row = {
+      id: broadcast.id,
+      name: broadcast.name,
+      updatedAt: new Date(broadcast.updatedAt).toISOString(),
+      triggeredAt: broadcast.triggeredAt
+        ? new Date(broadcast.triggeredAt).toISOString()
+        : "Not Triggered",
+    };
+    broadcastsRow.push(row);
+  });
+
+  return (
+    <DataGrid
+      rows={broadcastsRow}
+      sx={{
+        height: "100%",
+        width: "100%",
+        ".MuiDataGrid-row:first-child": {
+          borderTop: "1px solid lightgray",
+        },
+        ".MuiDataGrid-row": {
+          borderBottom: "1px solid lightgray",
+          paddingY: "0.5rem",
+        },
+        // disable cell selection style
+        ".MuiDataGrid-cell:focus": {
+          outline: "none",
+        },
+        // pointer cursor on ALL rows
+        "& .MuiDataGrid-row:hover": {
+          cursor: "pointer",
+        },
+      }}
+      getRowId={(row) => row.id}
+      onRowClick={(params) => {
+        router.push({
```

Comment:
Please use `<Link>`'s here instead of `router.push`!

Reply:
There is no direct way to wrap the entire row inside a `<Link>` tag. I would need to put the Link tag in every cell's `renderCell` property. Is that the right way to do it?

Reply:
hey @Wr4th100 yah, you can put it inside of the `renderCell` of the `baseColumn`. apologies for not responding sooner. I missed this comment!